src/thrift-client/TgtInterfaceImpl.cpp

```diff
@@ -828,19 +910,33 @@ void StordVmdk::ScheduleNow(folly::EventBase* basep, StorRpcAsyncClient* clientp
 
 	basep->runInEventBaseThread([this, clientp, basep,
 			pending = std::move(pending)] () {
+		std::unique_ptr<std::vector<::hyc_thrift::WriteRequest>> write;
+		uint32_t nwrites = 0;
+
 		for (auto reqp : pending) {
 			switch (reqp->type) {
 			case Request::Type::kRead:
 				ScheduleRead(basep, clientp, reqp);
 				break;
-			case Request::Type::kWrite:
-				ScheduleWrite(basep, clientp, reqp);
+			case Request::Type::kWrite: {
+				if (nwrites++ == 0) {
+					write = std::make_unique<std::vector<::hyc_thrift::WriteRequest>>();
+				}
+				auto data = std::make_unique<folly::IOBuf>
+					(folly::IOBuf::WRAP_BUFFER, reqp->bufferp, reqp->buf_sz);
+				write->emplace_back(apache::thrift::FragileConstructor(),
```

Comment:
FragileConstructor is a normal C++ construct or thrift baesd ? If thrift one then may be it make sense to replace with a get method instead of using it direclty ?